### DISCUSSION TITLE
Sign-in button capabilities like in Googles Demo

### DISCUSSION THREAD

**Discussion by BananaAcid** [MasterKale/SimpleWebAuthn#579] [General]
## Describe the issue

I try to implement a sign-in button like on [Google's Passkey Demo](https://passkeys-demo.appspot.com/one-button) 

- after working through the example and reading the docs, I am not able to spot any Info about it -- It looks to me like this project is only for protecting a successful login with a 2FA by passkeys.

Is my conclusion actually correct ?

## Documentation 

An info in the Readme.md about this being possible or not would be great. 

## Additional context

The example's code for route "/generate-authentication-options" it states "You need to know the user by this point" [index.ts#L221-L222](https://github.com/MasterKale/SimpleWebAuthn/blob/master/example/index.ts#L221-L222)

Same for the "/verify-authentication" route. 

... for me, the purpose using sign-in with passkeys is, not having the user logged in by that point ;)

**Comment by MasterKale** [comment]
There's a couple sections in the docs that might help shed some light on how to pull off an experience like the one on Google's demo site:

- **Advanced Guides > Passkeys > generateAuthenticationOptions()**: https://simplewebauthn.dev/docs/advanced/passkeys#generateauthenticationoptions
- **Packages > @simplewebauthn/browser > startAuthentication() > Browser autofill, a.k.a. Conditional UI**: https://simplewebauthn.dev/docs/packages/browser#browser-autofill-aka-conditional-ui

Take a look at those and let me know if that helps shed some light on this.

tl;dr: Make sure `allowCredentials` is an empty array before options make it to `navigator.credentials.get()`, either manually when calling `generateAuthenticationOptions()` or by calling `startAuthentication(..., true)` when using conditional UI.
